samples/azure/azure-service-fabric-routing/sample.md

```diff
@@ -0,0 +1,202 @@
+
+A partitioned endpoint can be configured to check that an incoming message should be processed locally. If it is not the case, the message is forwarded to the correct partition.
+
+### Message headers inspection
+
+Every incoming message has its `partition-key` header value inspected by the `DistributeMessagesBasedOnHeader` behavior. If the value specified in the header is matching the receiver`s partition key, then the message is processed. Otherwise, the message is forwarded to the appropriate partition specified by the header value. If the partition key is wrongly assigned, or the specified partition does not exist, the message is moved to the error queue.
+
+If the `partition-key` header does not exist, the pipeline execution continues to the *Message body inspection* step.
+
+### Message body inspection
+
+If message headers inspection is unsuccessful with value extraction of `partition-key` header, then the message body is used to determine the partition key value. The `DistributeMessagesBasedOnPayload` behavior determines the partition value using the mapping function provided via the configuration API. The result of the mapping function is stored as the `partition-key` header value. The inspection logic can raise a `PartitionMappingFailedException` exception if the partition key is not a valid partition key for a given endpoint. For a partition key that is not local, the message is forwarded to the appropriate partition. 
+
+The forwarding/processing decision is made in the same way as in the *Header inspection* step.
+
+NOTE: `PartitionMappingFailedException` is considered an unrecoverable exception. Whenever such an exception is raised the message that triggered the exception will be moved to the configured error queue. For more information refer to [unrecoverable exceptions documentation](/nservicebus/recoverability/custom-recoverability-policy.md).
+
+### Control message forwarding
+
+When an endpoint instance receives a control message representing either [Subscribe or Unsubscribe intent](/nservicebus/messaging/headers.md#messaging-interaction-headers-nservicebus-messageintent), the message is forwarded to all other partitions.
+
+### Configuration
+
+To enable Receiver Side Distribution for a specific endpoint, two arguments need to be provided:
+- Endpoint discriminators that are based on Service Fabric partition keys 
+- Mapping function per incoming message type that maps incoming messages to a partition key
+
+To enable Server Side Distribution, the following code is involved:
+
+snippet: ConfigureReceiverSideDistribution-CandidateVoteCount
+
+In many low-throughput scenarios Receiver Side Distribution might be enough to get started. To achieve better performance, it would be better to route directly to the partitions from the sender. See *Sender Side Distribution* below and how it can be used in combination with Receiver Side Distribution.
```

Comment:
> it would be better to route directly to the partitions from the sender. See *Sender Side Distribution* below and how it can be used in combination with Receiver Side Distribution.

SSD is **immediately** next so probably better to say:

it would be more efficient to route directly to the partitions from the sender with Sender-Side Distribution.